### PR TITLE
Ch1 Typos and Refactors

### LINKED ISSUE #3: feedback on ch 1 (2025-aug-18 version)
High level comments
---------------------

Sec 1.3.2 (empirical) mixes models for supervised and unsupervised
learning, whereas sec 1.3.1 (analytical) is almost entirely focused
on unsupervised representaion learning (ie., linear autoencoders x->z->xhat
with various constraints on z). Maybe you should add a little section on
supervised methods (eg linear and logistic regression) to 1.3.1 to make things more consistent?
You can then introduce the concept of a task and a loss function,
and say that the goal of supervised learning is just
prediction (potentially via a bottleneck which may or may not
recover the low-dim manifold), whereas the goal of unupervised is to
explicitly discover the manifold (which can then be used for prediction etc).
Then explain that when the relevant functions are linear, one can often solve
analytically, but when nonlinear, empirical methods have been favored.
(Actually this overall framing could make a nice intro sec 1.3.0,
before going into the analytical/empirical (or linear/nonlinear)
distinction in 1.3.1 and 1.3.2)

Also, the overall writing style is rather informal, and seems
more similar to a talk or blog post than a textbook.
It does not have the level of rigor as following chapters.
Also, there are too many foonotes, and lots of drive-by remarks to ideas and
concepts that are not defined (or even given citations) that many
readers may not understand. You are at risk of losing readers
who may not know these terms (even though most are either unneeded,
or will presumably/hopefully be defined later).

Finally, it seems that 1.4 was written separately from 1.2 and 1.3,
and there are no cross references between them. Sec 1.4 seems more polished.
Perhaps all sections could be unified better?
I suggest:
1.3.0 (or 1.2) defines the tasks of supervised/unsupervised,
and introduces concepts like cross entropy loss, reconstruction loss,
discriminator loss, and the need for constraints on the latent code z.
1.3.1 studies linear models for these tasks analytically.
1.3.2 studies nonlinear models for these tasks empirically.
1.4 is the new content, which is analytical approaches to nonlinear models.

----

Low-level comments (mostly typos)

p5. "Wiender had described.". Typo.

p12. eq 1.3.1. * operator (=convolution?) not defined.

p15. "folks like Erkki Oja". Too informal.

p15. Fig 1.11. Expand caption to explain why PCA and ICA
directions differ so much.

p16. Eq 1.3.18. Maybe better to write
"x ->[enc]-> z -> [A*] -> xhat",
where A* is the known linear decoder?
This makes the connection with DL in 1.3.21 clearer.
And maybe add vertical arrow into z to indicate
the sparsity constraint (eg with strength lambda)?

p17. It's a bit odd to characterize HJA20 (the DDPM paper)
as "image denoising" (which suggest going from fig 1.8b to 1.8c).
I would say it is "image generation from noise" (going from 1.8a to 1.8c)?

p18. Re eqn 1.3.26, it is very counterintuitive to think
of diffusion generative models as compressing noise samples
by mapping them to data samples, since we usually think about
compressing data samples. Maybe worth commenting on.

p20. "It is known that most of our brain is dedicated to process visual
information". Citation?  For which animals is this true?

p22. "It is believed that nature does not learn by back propagation, because
such a mechanism is still way too expensive for physical
implementation by nature". I think the reason it is unnatural is that
it requires global propagation of a signal (the gradient of the
final scalar loss) through all layers. Maybe cite some work on local
learning? And maybe mention that in this book, you don't care about
such biological constraints, and use BP freely (IIUC)?

p23. "as it surpassed extent classification methods ".
Extant? Existing?

p24. "Deep networks were later
adopted by the DeepMind team, led by Demis Hassabis,
to learn decision-making or control policies for playing games."
This ignores many years of prior pioneering work from Sutton, Barto
(who won 2024 Turing Award - worth mentioning!)
on (shallow) neural nets for RL.

p24. "The information bottleneck
framework [TZ15] was proposed in 2015 to analyze the role of deep
networks in such a context."
Not exactly. IB is a different model of the form X->Z->Y
with stochastic Z and explicit MI regularizers on the two edges.
This is not an analysis of vanilla deterministic X->Y mappings
trained only with prediction loss.

p24, Eq 1.3.32. Please don't use Z for observed labels! use Y.

p25. "z normally represents the texts that describe the
content of a desired image x. "
This may be confusing. Perhaps say that z is noise (just like in PCA)
and we want to learn a conditional mapping p(xhat|z,y)
where y=apple (say). We train this via x->z->xhat<-y
with loss in 1.3.34 augmented with conditioning signal y (which
we can then drop from notation).
This makes it more similar to autoencoders in sec 1.3.1.

p25. Fig 1.19. Policy and value functions are not defined.
May be unclear what the picture  represents.

p25. "Intuitively, the harder we could separate
X and X, probably the closer they are."
Not fluent. Also eq 1.3.35  is unclear.

p26. in eq 1.3.36, Is d the discriminator or the distance function
in eq 1.3.34? (Maybe use l in 1.3.34 instead?)

p27. "certain seemingly necessary components (e.g.,
dropout, normalization, attention, etc.) or their overall behaviors
(e.g., double descent, neural collapse" Lots of undefined buzzwords!
Need references.

p29. Eq 1.4.5. Explain that L is coding length (number of balls used by Enc(S)).
Also, you need to remind readers how f (on LHS) connects to S (on RHS),
since in unsupervised learning, it is more common to think
of samples Si as vectors coming from a joint distribtuion on R^D,
rather than as seequences being autoregressively generated by a function f.

p29. Eqn 1.4.7 Are we minizing rate wrt Encoder for known f,
or  are we joinlty minimizing rate wrt Encoder and f,
or is Encoder determined by f (somehow)?
Maybe write
min_f E[L(Enc(Si|f)) | Si~f(Data)]
  st ||Dec(Enc(Si|f)) - Si|_2^2 < eps
However, the epsilon ball loss on Si seems to be at odds with the arguments
against using L2 distance to measure d(X, Xhat) for images.
Please explain.

p30. Sec 1.4.2. Re "informative representation".
Maybe cite and discuss V-information from

  Y. Xu, S. Zhao, J. Song, R. Stewart, and S. Ermon, 
"A Theory of Usable Information Under Computational Constraints,” 
in NIPS, Feb. 2020. Available: http://arxiv.org/abs/2002.10689
  

p32. Fig 1.23 is nice, should be introduced much earlier
(eg in sec 1.3.0, where you introduce the idea of an autoencoder,
before discusing specific cases like PCA).

p33. "Note that in nature, animals are capable of learning
all by themselves without comparing
their estimate Xhat with the ground truth X in the data space.
They typically do not even have that option."
Well, if X is the observed input and Xhat is its reconstruction,
they can compare to ground truth. The issue is that L2 loss in pixel
space is dumb. (Reference the section on GANs)

p33. In eqn 1.4.17, you need to explain how  you avoid
avoid "latent collapse", in which E(X)=0 for all X,
which would result in Z=Zhat=0 and hence minimal loss
(as happens with other self-prediction methods, like SPR (and maybe DINO and JEPA?),
that need nasty stop gradients to prevent this pathology.)
Clarify that the encoder theta is trying to *maximize* L
(while only the decoder eta tries to minimize it),
which prevents this cheating!

### PR DESCRIPTION
**Pull request by DruvPai**
Went through the whole chapter 1 and fixed all writing to be grammatically correct. Some main contributions:
1) Artificial intelligence is the _practice_ of building machines which have _abstractive intelligence_ (as opposed to just being another name for abstractive intelligence itself). This also warrants a change for Ch8.
2) Removed information bottleneck as it was a slightly misleading context.
3) Added references to many facts which were not originally sourced, as well as references suggested in other PRs.
4) Fixed notation, including notation overload for `d` in the same paragraph.

This PR should resolve #3 and #17 .

### CODE REVIEW

**Comment by DruvPai** [comment]
Updated Ch8 with the same typo fixes, since it was already updated earlier in this same PR. This should resolve #13 .
